### PR TITLE
feat: add dataset base model & several datesets model

### PR DESCRIPTION
**Pull request by ChaosCodes**

### CODE REVIEW

File: beta_rec/datasets/dataset_base.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import numpy as np
+
+from beta_rec.utils.constants import *
+from beta_rec.utils.download import download_file, get_format
+
+
+class DatasetBase(object):
+    def __init__(self, url, dataset_name, dataset_file_name=None, download_filename=None, manual_download_url=None):
+        """Dataset base that any other datasets need to inherit from
+        This is an beta dataset which can derive to other dataset.
```
Comment (leungyukshing) [inline beta_rec/datasets/dataset_base.py:13]: should insert a newline

File: beta_rec/datasets/dataset_base.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        Several directory that store the dataset file would be created in the initial process.
+        Args:
```
Comment (leungyukshing) [inline beta_rec/datasets/dataset_base.py:15]: what about the args `manual_download_url`?

File: beta_rec/datasets/dataset_base.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import shutil
+import pandas as pd
+import numpy as np
+
+from beta_rec.utils.constants import *
+from beta_rec.utils.download import download_file, get_format
+
+
+class DatasetBase(object):
+    def __init__(self, url, dataset_name, dataset_file_name=None, download_filename=None, manual_download_url=None):
```
Comment (leungyukshing) [inline beta_rec/datasets/dataset_base.py:11]: why need manual_download_url?

File: beta_rec/datasets/epinions.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import sys
+import os
+import pandas as pd
+
+from beta_rec.utils.constants import *
+from beta_rec.datasets.dataset_base import DatasetBase
+
+epinions_url = 'http://www.trustlet.org/datasets/downloaded_epinions/ratings_data.txt.bz2'
```
Comment (leungyukshing) [inline beta_rec/datasets/epinions.py:8]: whether it would be better if we store these constants in a separate file?

Comment (mengzaiqiao) [reply beta_rec/datasets/epinions.py:8]: Agree. import * is not a good coding format. I have raised an issue for this.

**Pull request merged**